src/mscorlib/src/System/IO/Stream.cs

```diff
@@ -1059,6 +1059,8 @@ internal sealed class SyncStream : Stream, IDisposable
             private bool? _overridesBeginRead;
             [NonSerialized]
             private bool? _overridesBeginWrite;
+            [NonSerialized]
+            private readonly object _syncRoot = new object();
```

Comment:
With this marked [NonSerialized], won't _syncRoot be left uninitialized and null when such a SyncStream is deserialized, causing NullReferenceExceptions on subsequent use?